internal/juju/secrets.go

```diff
@@ -238,3 +258,57 @@ func (c *secretsClient) DeleteSecret(input *DeleteSecretInput) error {
 
 	return nil
 }
+
+// GrantSecretAccess grants access to a secret.
+func (c *secretsClient) GrantSecretAccess(input *GrantRevokeSecretAccessInput) error {
+	conn, err := c.GetConnection(&input.ModelName)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = conn.Close() }()
+
+	secretAPIClient := c.getSecretAPIClient(conn)
+
+	secretURI, err := coresecrets.ParseURI(input.SecretId)
+	if err != nil {
+		return err
+	}
+
+	// In the current implementation, the secret name is not used.
+	results, err := secretAPIClient.GrantSecret(secretURI, "", input.Applications)
+	if err != nil {
+		return typedError(err)
+	}
+	if results[0] != nil && len(results) > 1 {
```

Comment:
This is a bit confusing to read, does any result being returned mean that we have an error?

Maybe put a comment here to explain whats happening

Reply:
thx, Caner. That's good idea. I'll move it to separate func.